Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.cpp

```diff
@@ -96,6 +96,11 @@ void WebSharedWorkerContextManagerConnection::updatePreferencesStore(const WebPr
 {
     WebPage::updatePreferencesGenerated(store);
     m_preferencesStore = store;
+
+#if USE(AUDIO_SESSION)
+    if (m_preferencesStore->getBoolValueForKey(WebPreferencesKey::mediaPlaybackEnabledKey()))
+        AudioSession::sharedSession().setEnabled(true);
+#endif
```

Comment:
I don't think it makes sense to enable/disable the audio session do this here. Is there a reason it isn't done in `WebPage::updatePreferencesGenerated`?

Reply:
Not really, I could add it to `WebPage::updatePreferencesGenerated`.